examples/transitions/main.tf

```diff
@@ -0,0 +1,39 @@
+#
+# Private Bucket with transitions
+#
+
+module "s3_private_bucket" {
+  source = "../../"
+
+  bucket                   = var.test_name
+  use_account_alias_prefix = false
+
+  abort_incomplete_multipart_upload_days = 7
+
+  transitions = [
+    {
+      days          = 30
+      storage_class = "STANDARD_IA"
+    },
+    {
+      days          = 60
+      storage_class = "GLACIER"
+    },
+    {
+      days          = 150
+      storage_class = "DEEP_ARCHIVE"
```

Comment:
I should note that moving things to deep archive requires that `days` is 90 days larger than whatever you have for Glacier.